openedx_external_enrollments/external_enrollments.py

```diff
@@ -0,0 +1,424 @@
+"""External enrollments file."""
+import datetime
+import logging
+from collections import OrderedDict
+
+import requests
+from django.conf import settings
+from oauthlib.oauth2 import BackendApplicationClient
+from opaque_keys.edx.keys import CourseKey
+from requests_oauthlib import OAuth2Session
+from rest_framework import status
+
+from openedx_external_enrollments.edxapp_wrapper.get_courseware import get_course_by_id
+from openedx_external_enrollments.edxapp_wrapper.get_student import CourseEnrollment, get_user
+from openedx_external_enrollments.models import (
+    EnrollmentRequestLog,
+    ProgramSalesforceEnrollment,
+)
+
+LOG = logging.getLogger(__name__)
+
+
+class BaseExternalEnrollment(object):
+    """
```

Comment:
Docstrings throughout the file... @andrey-canon

Reply:
out of the scope of this pr